### PR TITLE
introduced DataSort class

### PR DESCRIPTION
**Pull request by alicak**
Added DataSort class that supports sorting the way DataFilter supports filtering.

### CODE REVIEW

File: api/src/main/java/io/lumeer/engine/api/data/DataStorageDialect.java [hunk 1]
```diff
@@ -34,6 +34,8 @@
 
    DataDocument addRecentlyUsedDocumentQuery(final String metadataCollection, final String collection, final String id, final int listSize);
 
+   String sortStringForSearch(final String field, final int order); // produces string that can be used in dataStorage.search() method
+
    // LinkingFacade
```
Comment (kubedo8) [inline api/src/main/java/io/lumeer/engine/api/data/DataStorageDialect.java]: This method is unused now, istn't it? (And probably unwanted)

Comment (alicak) [inline api/src/main/java/io/lumeer/engine/api/data/DataStorageDialect.java]: Yes, you're right.

File: mongodb-storage/src/main/java/io/lumeer/mongodb/MongoDbStorageDialect.java [hunk 2]
```diff
@@ -67,6 +69,10 @@ public DataDocument addRecentlyUsedDocumentQuery(final String metadataCollection
                               .append("$slice", listSize))));
    }
 
+   public String sortStringForSearch(final String field, final int order) {
+      return MongoUtils.convertBsonToJson(Sorts.descending(field));
```
Comment (livthomas) [inline mongodb-storage/src/main/java/io/lumeer/mongodb/MongoDbStorageDialect.java]: What is this method good for? I cannot see it used anywhere and it seems to ignore `order` parameter.

Comment (alicak) [inline mongodb-storage/src/main/java/io/lumeer/mongodb/MongoDbStorageDialect.java]: I'm sorry I didn't check it. It is forgotten method and I have already removed it.

**Pull request merged**